spec/controllers/feedbacks_controller_spec.rb

```diff
@@ -0,0 +1,28 @@
+require "rails_helper"
+
+RSpec.describe FeedbacksController, type: :controller do
+  describe "#create" do
+    before do
+      allow(controller).to receive(:level_of_help).and_return("controlled")
+    end
+
+    it "saves freetext feedback correctly" do
+      post :create, params: { type: "freetext", text: "foo", page: "bar" }
+      expect(FreetextFeedback.find_by(text: "foo", page: "bar", level_of_help: "controlled")).not_to be_nil
+    end
+
+    it "saves satisfaction feedback correctly" do
+      allow(controller).to receive(:outcome).and_return("eligible")
```

Comment:
Likewise here, we're not really checking that our logic to retrieve data from the session is working correctly. Better to have some session data and an assessment code param.

Reply:
to chat about when pairing